### ISSUE TITLE
[FR] Self-hosted : bundle web app in Cloud distribution

### ISSUE THREAD

**Issue by almereyda** [AppFlowy-IO/AppFlowy-Cloud#873]
**Describe the bug**

The release of v0.7.1 has seen the ascent of a feature to collaborate on the web. With the frontend in the apps and for a local build needing to adapt to a self-hosted AppFlowy-Cloud instance at runtime https://github.com/AppFlowy-IO/AppFlowy/issues/6539 comes also the need, that the web app is bundled with the Cloud distribution.

Right now for #680 it is possible to work around the present limitations in the client https://github.com/AppFlowy-IO/AppFlowy/issues/5920 with building a heavily monkey-patched app that is able to show the static content of a published page.

Ideally the manual build-time modifications aren't necessary and AppFlowy-Cloud comes with a web app (build pipeline), that allows to adapt to the URL of the AppFlowy instance it is published with.

**To Reproduce**

1. Open the AppFlowy App
2. Click on Invite to collaborate and copy the link.
3. Paste the link into the browser's address bar and replace appflowy.com with your custom URL of the web frontend
4. Click on one of the enabled authentication providers and be redirected into the app.

**Expected behavior**

AppFlowy opens in the browser and it is possible to edit the document.

**Desktop (please complete the following information):**
 - OS: Flatpak
 - Browser Firefox
 - Version v0.7.1
 
### Additional Context

We have four highly related tracking issues for two features in two projects now:

Feature | AppFlowy | AppFlowy Cloud
--- | --- | ---
Share to web | https://github.com/AppFlowy-IO/AppFlowy/issues/5920 | https://github.com/AppFlowy-IO/AppFlowy-Cloud/issues/680
Collaborate on web | https://github.com/AppFlowy-IO/AppFlowy/issues/6539 | \<this one\>

**Comment by khorshuheng** [comment]
Yes, this is indeed a popular request. For the first step, i will need to setup the github action to build and publish the docker image for appflowy web. After that i will add the additional service to docker compose.

Changes to AppFlowy Flutter app to provide a setting for custom appflowy web url will come later, though, as that will require additional effort.

**Comment by almereyda** [comment]
There are mare aspects, which can be considered for dynamic configuration of the Flutter app:

- [ ] https://github.com/AppFlowy-IO/AppFlowy-Cloud/issues/565  
  There are several places where AI functionalities are offered for self-hosted setups, even if they were not started and no OpenAI API Key is given.

Also I would like to continue to encourage a less "inviting" (others say "clickbait") stance for published pages, providing less/no sign up or log in links. https://github.com/AppFlowy-IO/AppFlowy/issues/5920#issuecomment-2409106343

- [ ] https://github.com/AppFlowy-IO/AppFlowy-Cloud/issues/621  
  Should maybe live in AppFlowy-IO/AppFlowy instead of here now, since it now requires the `web_app` and as such the Flutter App to adapt to available authentication options, when using a self-hosted AppFlowy-Cloud instance.

**Comment by almereyda** [comment]
With the recent `web` related commits in this repository, a separate AppFlowy-Web repository has emerged. It contains a Dockerfile for a container which is not yet published to Docker Hub (nor the GitHub Container Registry at ghcr.io). The instructions for the image do not yet give any indication, if or how the custom domain name of a self-hosted instance is to be made recognised by the web app. Eventually it just works?

We'd need to run it and reproduce, whether the found artefacts and build instructions help to replace the earlier manual build of a patched web frontend https://github.com/AppFlowy-IO/AppFlowy-Cloud/issues/680#issuecomment-2380387584.

**Comment by khorshuheng** [comment]
As the endpoint  for AppFlowy Cloud is determined during the time the image is built (via .env), as opposed to run time variable, we are not able to create a Appflowy Web docker image that can fit every deployment. So currently, the recommended approach is to set your Appflowy cloud URL in the .env file for AppFlowy Web repo, then use the makefile to build an image, which you can then deploy.

That being said, we are also planning to release an official image for Appflowy web, that is configured to work with Appflowy cloud deployed on localhost (which can be set up via docker compose up), so that it's easier for user  to evaluate AppFlowy Web via a reproducible set up.

**Comment by khorshuheng** [comment]
There are improvements needed, such as ability of building the image without pnpm being installed on host machine, simplifying the installation process, and better documentation. But yes , it is possible to deploy the web app as described.

**Comment by bdruth** [comment]
@wx-11 I am clearly missing something, but if I pull `main`, the `Makefile` has no target to make `image`
```
❯ git pull
Already up to date.
❯ make image
make: *** No rule to make target 'image'.  Stop.
```

**Comment by luxio** [comment]
> @wx-11 I am clearly missing something, but if I pull `main`, the `Makefile` has no target to make `image`

I think the you should run `make image` within the [AppFlowy-Web](https://github.com/AppFlowy-IO/AppFlowy-Web) repository.

**Comment by bdruth** [comment]
@luxio - ty! Didn't realize there was another repo, much obliged.

**Comment by khorshuheng** [comment]
Since https://github.com/AppFlowy-IO/AppFlowy-Cloud/pull/1196 , AppFlowy Web is now bundled together with the docker compose file. For fresh installation, AppFlowy Web should work out of the box, and doesn't require building new images, nor cloning the AppFlowy-Web repository.

In addition, there is now an official AppFlowy Web image (appflowyinc/appflowy_web).

For existing self hosters, nginx.conf, docker compose file, and .env files will need to be updated based on the current main branch.

**Comment by khorshuheng** [comment]
Proposal in https://github.com/AppFlowy-IO/AppFlowy-Cloud/issues/873#issuecomment-2440046910 have not been implemented yet, though i think we can create new issues to track them. I will close the current issue.
